packages/components/src/components/file-dropzone/file-dropzone.tsx

```diff
@@ -1,11 +1,43 @@
 import classNames from 'classnames';
-import React from 'react';
+import React, { ReactElement, ReactNode, RefObject } from 'react';
 import { CSSTransition } from 'react-transition-group';
-import Dropzone from 'react-dropzone';
+import Dropzone, { DropzoneOptions } from 'react-dropzone';
 import { truncateFileName } from '@deriv/shared';
 import Text from '../text';
 
-const FadeInMessage = ({ is_visible, color, children, key, timeout, no_text }) => (
+type TFadeInMessage = {
+    is_visible: boolean;
+    color?: string;
+    key?: string;
+    timeout: number;
+    no_text?: boolean;
+};
+
+type TPreviewSingle = {
+    dropzone_ref: RefObject<HTMLElement>;
+} & TFileDropzone;
+
+type TFileDropzone = {
+    className?: string;
+    noClick?: boolean;
+    validation_error_message: (open?: () => void) => ReactNode | null;
+    max_size?: number;
+    value: Array<{ name: string }>;
+    message: (open?: () => void) => ReactNode;
```

Comment:
it is used both as a string and as a callback inside our component, so I suggest this:

```suggestion
    message: ReactNode | (open?: () => void) => ReactNode;
```